docs/backend/indexing.md

```diff
@@ -0,0 +1,212 @@
+        <field name="specialfield" type="zope.schema.TextLine"
+                indexer:searchable="true">
+        <title>Special field</title>
+        </field>
+
+    </schema>
+</model>
+```
+
+
+Your SearchableText indexer includes now your custom field on your behavior. 
+It is enabled on your content type, if the `plone.textindexer` behavior is enabled there too.
+
+
+#### Registering a custom field converter
+
+
+By default, a field is converted to a searchable text by rendering the widget in display mode and transforming the result to `text/plain`. 
+However, if you need to convert your custom field in a different way, you only have to provide a more specific converter multi-adapter.
+
+Convert multi-adapter specification:
+
+:Interface: `plone.app.dexterity.textindexer.IDexterityTextIndexFieldConverter`
+:Discriminators: context, field, widget
+
+Example:
+
+```python
+from plone.app.dexterity.textindexer.converters import DefaultDexterityTextIndexFieldConverter
+from plone.app.dexterity.textindexer.interfaces import IDexterityTextIndexFieldConverter
+from my.package.interfaces import IMyFancyField
+from plone.dexterity.interfaces import IDexterityContent
+from z3c.form.interfaces import IWidget
+from zope.component import adapter
+from zope.interface import implementer
+
+@implementer(IDexterityTextIndexFieldConverter)
+@adapter(IDexterityContent, IMyFancyField, IWidget)
+class CustomFieldConverter(DefaultDexterityTextIndexFieldConverter):
+
+    def convert(self):
+            # implement your custom converter
+            # which returns a string at the end
+            return ''
+```
+
+ZCML:
+
+```xml
+<configure xmlns="http://namespaces.zope.org/zope">
+
+    <adapter factory=".converters.CustomFieldConverter" />
+
+</configure>
+```
+
+There is already an adapter for converting files properly. 
+
+
+
+#### Extending indexed data
+
+
+Sometimes you need to extend the SearchableText with additional data which is not stored in a field. 
+It is possible to register a named adapter which provides additional data:
+
+```python
+from plone.app.dexterity import textindexer
+from zope.component import adapter
+from zope.interface import implementer
+
+
+@implementer(textindexer.IDynamicTextIndexExtender)
+@adapter(IMyBehavior)
+class MySearchableTextExtender(object):
+
+    def __init__(self, context):
+        self.context = context
+
+    def __call__(self):
+        """Extend the searchable text with a custom string"""
+        return 'some more searchable words'
+```
+
+ZCML:
+
+```xml
+<configure xmlns="http://namespaces.zope.org/zope">
+
+    <adapter factory=".indexer.MySearchableTextExtender"
+                name="IMyBehavior"
+                />
+
+</configure>
+```
+
+This is a **named** adapter! This makes it possible to register multiple
+extenders for the same object on different behavior interfaces. The name of
+the adapter does not matter, but it's recommended to use the name of the
+behavior (this may reduce conflicts).
```

Comment:
This is a **named** adapter! 
The named registration allows to register multiple extenders on different behavior interfaces applying to the same object.
The name of the adapter does not matter, but it's recommended to use the name of the behavior (this may reduce conflicts).